DistroLauncher/console_service.h

```diff
@@ -63,17 +65,25 @@ namespace Win32Utils
             // the consumer side, both stdout and stderr goes to the same file.
             if (state == consoleState())
                 return;
-            auto res2 = _dup2(state.stdErrFileDescriptor, _fileno(stderrStream));
+            fflush(stderrStream);
+            fflush(stdoutStream);
             SetStdHandle(nStderrHandle, state.stdErrHandle);
-            auto res = _dup2(state.stdOutFileDescriptor, _fileno(stdoutStream));
             SetStdHandle(nStdoutHandle, state.stdOutHandle);
+            auto res2 = _dup2(state.stdErrFileDescriptor, _fileno(stderrStream));
+            auto res = _dup2(state.stdOutFileDescriptor, _fileno(stdoutStream));
         }
 
       public:
         // std::forward<Pipe>(pipe) is just a way to ensure pipe is moved instead of copied.
         ConsoleService(Pipe&& pipe) noexcept : redirectTo{std::forward<Pipe>(pipe)}
         {
             assert(redirectTo.readHandle() != nullptr);
+            // Attempting to find the console window is best-effort if it is not the old style console.
+            // This has been tested for the new Windows Terminal. Other terminals out there may not fit.
+            window_ = FindWindow(L"CASCADIA_HOSTING_WINDOW_CLASS", DistributionInfo::WindowTitle.c_str());
```

Comment:
hum, is this working with the default windows command line app? as ubuntu.exe will be launched by default with it.

Reply:
It depends on user settings. In my laptop everything that launches a console effectively runs on the new Windows Terminal. I'd expect from the majority of the users to be the old console. Very, very few should run something other than those two options (my guess based on the how easy is to access those alternatives, I don't have data to sustain that).

Reply:
Hum, I think I will let you decide. Indeed, ` U+200E` is too magic in my opinion. If the current approach is good enough, let’s keep it. If we see more corner cases, maybe we can reevaluate that? Anyway, that’s really up to you :)

Reply:
Alright. Thank you. I'll keep it off of the table for now.